### ISSUE TITLE
[Bug]: Allow for non-english characters where it should be allowed. 

### ISSUE THREAD

**Issue by erwangranger** [opendatahub-io/odh-dashboard#1968]
### ODH Component

ODH Dashboard

### Current Behavior

From https://issues.redhat.com/browse/RHODS-6346  and slack: https://redhat-internal.slack.com/archives/C03UGJY6Z1A/p1697115227770089 

In various fields, it's possible to type in some non-english characters, but it then becomes impossible to save. 

When saving, there is an error stating:

```
----
the object provided is unrecognized (must be of type Secret): 
couldn't get version/kind; json parse error: 
unexpected end of JSON input ({"apiVersion":"v1","kind":"Sec ...)
```

Some examples: 

<img width="756" alt="image" src="https://github.com/opendatahub-io/opendatahub-community/assets/32554862/8dc65f78-d7d1-40a1-a817-335cced549c9">

![image](https://github.com/opendatahub-io/opendatahub-community/assets/32554862/0ba37d8d-6b74-4193-8eef-5ce360cc7653)

### Expected Behavior

For each field where I can write text:
* Either the special characters are allowed, and they should be saved properly
* Or they are disallowed, and the application should clearly indicate it to the user, instead of failing to save.

### Steps To Reproduce

here are some sample characters to test with: 
```
à 
é
ł (unicode hex 0142) 
Ĥ (unicode hex 124)
```

### Open Data Hub Version

```yml
RHODS 1.33
```

**Comment by andrewballantyne** [comment]
It appears this is done in YAML editor in OpenShift Console pretty straight forward. That might be the monaco editor and that might be the OpenShift Console. I don't think this `needs-info` anymore. We can manage these values however they do. Looks to be some conversation of the string.

**Comment by manosnoam** [comment]
This has been reported since RHODS 1.20... https://issues.redhat.com/browse/RHODS-7329.
This is now fixed in RHODS  2.4: 
It is not allowed to enter unsupported Unicode characters in specified fields:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/9913945/b45cb5e4-6bbf-47ad-b960-417c235e4f28)

**Comment by manosnoam** [comment]
@andrewballantyne please note that this specific bug was resolved with a proper error message, but related bugs were not:
[RHODS-12901 - No input validation when updating existing resource](https://issues.redhat.com/browse/RHODS-12901)

[RHODS-6346 - Missing input validation in item creation forms/fields](https://issues.redhat.com/browse/RHODS-6346)
